pkg/util/http/client.go

```diff
@@ -31,86 +27,68 @@ func SanitizeURL(message string) string {
 	return apiKeyRegExp.ReplaceAllString(message, apiKeyReplacement)
 }
 
-// CreateHTTPTransport creates an *http.Transport for use in the agent
-func CreateHTTPTransport() *http.Transport {
-	tlsConfig := &tls.Config{
-		InsecureSkipVerify: config.Datadog.GetBool("skip_ssl_validation"),
+// Client wraps (http.Client).Do and resets the underlying connections at the
+// configured interval
+type Client struct {
+	httpClient    *http.Client
+	newHTTPClient func() *http.Client
+	resetInterval time.Duration
+
+	lastReset time.Time
+	mutex     sync.RWMutex
+}
+
+// NewClient returns an initialized Client resetting connections at the passed resetInterval.
+// The underlying http.Client used will be created using the passed http client factory.
+func NewClient(resetInterval time.Duration, newHTTPClient func() *http.Client) *Client {
+	return &Client{
+		resetInterval: resetInterval,
+		httpClient:    newHTTPClient(),
+		newHTTPClient: newHTTPClient,
+	}
+}
+
+// Do wraps (http.Client).Do. Thread safe.
+func (c *Client) Do(req *http.Request) (*http.Response, error) {
+	c.mutex.RLock()
+	httpClient := c.httpClient
+	c.mutex.RUnlock()
+	if c.shouldReset() {
```

Comment:
indeed, this case has a small perf hit, but will work overall: the "old" `httpClient` will in this case have no idle connections, so it will open a new one, which will only be used for this one  http request (hence the perf hit). Pushing a change that will make this edge case much less likely.